Spiegel Online.js

```diff
@@ -39,151 +39,104 @@ http://www.spiegel.de/international/europe/0,1518,700530,00.html
+	
+	// attachement
+	if (url.match(/^http\:\/\/www\.spiegel\.de\/spiegel/)){
 		var printurl_xPath = ".//div[contains(@class, 'article-function-box')]/ul/li[1]/a/@href";
 		var printurl = ZU.xpathText(doc, printurl_xPath);
 		//Zotero.debug(printurl);
-		newItem.attachments.push({url:printurl, title:doc.title, mimeType:"application/pdf"});
+		newItem.attachments.push({url:printurl, title:"Full Text PDF", mimeType:"application/pdf"});
 	} else { 
 		// Attachment. Difficult. They want something inserted into the URL.
-		var printurl = doc.location.href;
+		var printurl = url;
 		printurl = printurl.replace(/(\d+\,\d+\.html.*$)/, 'druck-$1'); //done!
-		newItem.attachments.push({url:printurl, title:doc.title, mimeType:"text/html"});
+		newItem.attachments.push({url:printurl, title:"Snapshot", mimeType:"text/html"});
 	}
 	
-	//Ausgabe/Volume für Print
+	// Ausgabe/Volume für Print
 	if (ZU.xpathText(doc, '//div[@class="spiegel-magazin-title asset-title"]') && newItem.itemType == "magazineArticle"){
 		newItem.volume = ZU.xpathText(doc, '//div[@class="spiegel-magazin-title asset-title"]').match(/(\d+)\/\d{4}/)[1];
 	}
 	
 	// Summary
-	var summary_xPath = ".//p[@class='article-intro']";
-	if (doc.evaluate(summary_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext() ){ 
-		var summary= doc.evaluate(summary_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		newItem.abstractNote = Zotero.Utilities.trim(summary);
+	var summary = ZU.xpathText(doc, './/p[@class="article-intro"]')
+	if (summary) { 
+		newItem.abstractNote = summary.trim();
 	}
 	
-	// Date - sometimes xpath1 doesn't yield anything. Fortunately, there's another possibility...
-	var date1_xPath = ".//h5[contains(@id, 'ShortDate')]"; 
-	var date2_xPath = "//meta[@name='date']";
-	var date3_xPath = "//div[@id='spShortDate']"
-	if (doc.evaluate(date1_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext() ){ 
-		var date= doc.evaluate(date1_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		if (date.match('/')) {
-			date = date.replace(/(\d\d)\/(\d\d)\/(\d\d\d\d)/, "$2.$1.$3").replace(/T.+/,"");
-		}
-	} else if (doc.evaluate(date2_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext() ){ 
-		var date= doc.evaluate(date2_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().content;
-		date=date.replace(/(\d\d\d\d)-(\d\d)-(\d\d)/, '$3.$2.$1').replace(/T.+/,"");
-	} else	if (doc.evaluate(date3_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext() ){ 
-		var date= doc.evaluate(date3_xPath, doc, null, XPathResult.ANY_TYPE, null).iterateNext().textContent;
+	// Date
+	var date = ZU.xpathText(doc, './/h5[contains(@id, "ShortDate")]')
+		|| ZU.xpathText(doc, '//meta[@name="date"]/@content')
+		|| ZU.xpathText(doc, '//div[@id="spShortDate"]');
+	if (date) {
+		date = date.replace(/(\d\d)\/(\d\d)\/(\d\d\d\d)/, "$3-$1-$2");
+		date = date.replace(/(\d\d)\.(\d\d)\.(\d\d\d\d)/, "$3-$2-$1");
+		date = date.replace(/T.+/,"");
```

Comment:
(though if there happens to be a "T" before the date, the whole date would removed, but I guess that probably doesn't happen)